compiler/qsc_rca/src/core.rs

```diff
@@ -405,11 +406,27 @@ impl<'a> Analyzer<'a> {
             callee_input_pattern_id,
             args_input_id,
             self.package_store,
+            fixed_args.as_ref().map_or(0, Vec::len),
         );
         let application_instance = self.get_current_application_instance();
 
         // Derive the compute kind based on the value kind of the arguments.
-        let arg_value_kinds = self.derive_arg_value_kinds(&arg_exprs);
+        let arg_value_kinds = if let Some(fixed_args) = fixed_args {
+            fixed_args
```

Comment:
For fixed arguments it would be worth adding a descriptive comment mentioning the assumptions (such as fixed arguments appearing first in the lifted lambda), and what `map` and `chain` are doing here.

Reply:
Added a comment for this.